File: backend/app/api/schemas/analysis.py
"""
Analysis API Schemas
"""

from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class AnalysisUploadRequest(BaseModel):
    """Request schema for video upload"""
    mentor_name: str = Field(..., min_length=2, max_length=100)
    subject: str = Field(..., min_length=2, max_length=100)
    mentor_id: Optional[str] = None


class ScoresResponse(BaseModel):
    """Scores response schema"""
    engagement: float
    communication: float
    technical_depth: float
    clarity: float
    interaction: float
    overall: float


class InsightsResponse(BaseModel):
    """Insights response schema"""
    strengths: List[str]
    improvements: List[str]
    recommendations: List[str]
    key_highlights: Optional[str] = None


class AnalysisStatusResponse(BaseModel):
    """Analysis status response"""
    analysis_id: str
    status: str
    message: Optional[str] = None
    created_at: datetime


class AnalysisResultResponse(BaseModel):
    """Complete analysis result response"""
    id: str
    mentor_id: str
    mentor_name: str
    subject: str
    video_filename: str
    video_duration: Optional[float] = None
    status: str
    scores: Optional[ScoresResponse] = None
    insights: Optional[InsightsResponse] = None
    transcript: Optional[str] = None
    created_at: datetime
    updated_at: datetime
    completed_at: Optional[datetime] = None


class AnalysisListResponse(BaseModel):
    """List of analyses response"""
    analyses: List[AnalysisResultResponse]
    total: int
    skip: int
    limit: int

File: frontendorg/src/types/analysis.ts
export interface Scores {
  engagement: number;
  communication: number;
  technical_depth: number;
  clarity: number;
  interaction: number;
  overall: number;
}

export interface Insights {
  strengths: string[];
  improvements: string[];
  recommendations: string[];
  key_highlights?: string;
}

export interface AudioFeatures {
  duration?: number;
  energy_mean?: number;
  energy_std?: number;
  pitch_mean?: number;
  pitch_std?: number;
  speech_rate?: number;
  pause_ratio?: number;
  sample_rate?: number;
}

export interface VisualFeatures {
  total_frames?: number;
  samples_analyzed?: number;
  gesture_count?: number;
  face_confidence?: number;
  face_presence_ratio?: number;
  eye_contact_ratio?: number;
  hand_gestures?: number;
}

export interface NLPAnalysis {
  technical_depth_score?: number;
  clarity_score?: number;
  structure_score?: number;
  question_count?: number;
  technical_term_count?: number;
  word_count?: number;
}

export type AnalysisStatus = 'pending' | 'processing' | 'completed' | 'failed';

export interface Analysis {
  id: string;
  mentor_id: string;
  mentor_name: string;
  subject: string;
  video_filename: string;
  video_duration?: number;
  status: AnalysisStatus;
  scores?: Scores;
  insights?: Insights;
  transcript?: string;
  audio_features?: AudioFeatures;
  visual_features?: VisualFeatures;
  nlp_analysis?: NLPAnalysis;
  created_at: string;
  updated_at: string;
  completed_at?: string;
  error?: string;
}

export interface AnalysisListResponse {
  analyses: Analysis[];
  total: number;
  skip: number;
  limit: number;
}

export interface UploadResponse {
  analysis_id: string;
  status: string;
  message: string;
}